### PR TITLE
Add eth-event-service

### PR DESCRIPTION
**Pull request by lsenta**
Implements the ETH Events Service that:
- subscribe to a list of contracts listed in the env,
- forward these events to a pub/sub topics to be processed by our backends (gcloud functions)

Adds Timestamp events to the ScriptHash Smart Contract.

### CODE REVIEW

**Comment by GrandSchtroumpf** [comment]
What is the problem with the linting ?

**Comment by lsenta** [comment]
> What is the problem with the linting ?

Console' logging is forbidden, which I think is ok: we can pick a logging library.

**Comment by GrandSchtroumpf** [comment]
> Console' logging is forbidden, which I think is ok: we can pick a logging library. I don't know if you have any favorites?

No favorite.

File: libs/script/src/lib/contracts/script-hash.ts [hunk 1]
```diff
@@ -27,3 +29,13 @@ export class ScriptHashService extends Contract {
     super(address, abi, provider.getSigner());
   }
 }
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ScripHashServiceReadOnly extends Contract {
+  constructor(provider: BaseProvider) {
```
[the remark this answers is not in the document]
Comment (GrandSchtroumpf) [reply libs/script/src/lib/contracts/script-hash.ts]: But this works only if you know in which environment you are. As this is in a library we don't know the link the envrionment file of the app....

File: apps/eth-events-service/src/main.ts [hunk 2]
```diff
@@ -18,3 +19,48 @@ app.listen(port, (err) => {
...
+      const contract = new Contract(addr, [], provider);
+
+      contract.on('*', event => {
+        topic.publish(Buffer.from(JSON.stringify(event)))
+          .then(r => console.info('[PUBLISHED]', 'contract:', name, 'topic:', topicName, 'event:', event, 'result:', r))
+          .catch(err => console.error('[ERROR]', 'contract:', name, 'topic:', topicName, 'error:', err));
+      });
+
+      return {
+        addr, name, topicName, topic, contract
```
[the remark this answers is not in the document]
Comment (lsenta) [reply apps/eth-events-service/src/main.ts]: Keep it around for when we need to manage the service, like pulling contracts update and restarting contracts, etc.

**Pull request merged**